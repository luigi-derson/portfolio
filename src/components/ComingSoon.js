import React from 'react'

const ComingSoon = () => {
  return (
    <div>
      <h2>&gt; Comitting changes...</h2>
      <p>Building a new visual experience</p>
    </div>
  )
}

export default ComingSoon
