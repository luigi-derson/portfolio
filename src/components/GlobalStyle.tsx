import { createGlobalStyle } from '@xstyled/styled-components'

const GlobalStyles = createGlobalStyle`
html,
body,
#___gatsby,
#gatsby-focus-wrapper {
  height: 100%;
}
body {
  font-family: sans;
  line-height: 1.4;
  color: text;
  background-color: background;
  transition: background-color 500ms cubic-bezier(0.33, 1, 0.68, 1), color 300ms ease;
}

h1, h2, h3, h4, h5, h6 {
  color: headings;
}

a {
  color: links;
}

.mesh-gradient-image {
  position: fixed;
  height: 100%;
  width: 100%;
  top: 0;
  left: 0;
  z-index: -1;
  opacity: 0.65;
  transition: opacity 1000ms cubic-bezier(0.12, 0, 0.39, 0);
}

.mesh-gradient-image--dark {
  transition: opacity 300ms cubic-bezier(0.12, 0, 0.39, 0);
  opacity: 0;
}
`

export default GlobalStyles
