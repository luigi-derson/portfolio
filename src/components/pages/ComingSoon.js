import React from 'react'
import styled from 'styled-components'
import { MarkGithubIcon, MailIcon } from '@primer/styled-octicons'

import { H2 } from 'components/atoms/headings'
import { Box, FlexBox } from 'components/atoms/box'

import logo from 'assets/luigi-derson_logo.svg'
import bgimage from 'assets/pawel-czerwinski-GHLapgsw0dg-unsplash-optimized.jpg'
import Text from 'components/atoms/text'
import Paragraph from 'components/atoms/paragraph'

const FOOTER_HEIGHT = `${80 / 16}em`

const ComingSoon = () => {
  return (
    <>
      <Box
        width={1}
        height="100%"
        position="absolute"
        pos="0"
        overflow="hidden"
        textAlign="center"
        bg="black"
        zIndex="-1"
      >
        <BgImg src={bgimage} />
      </Box>
      <Section>
        <FlexBox
          flexDirection="column"
          justifyContent="center"
          alignItems="center"
          height="100%"
        >
          <Box my={4}>
            <img height="60px" src={logo} alt="Luigi Derson Logo" />
          </Box>
          <H2 color="green" fontFamily="mono" fontWeight="regular" mb="4">
            &gt; Comitting changes...
          </H2>
          <Paragraph>Building a new visual experience</Paragraph>
        </FlexBox>
      </Section>
      <Footer
        as="footer"
        textAlign="center"
        flexDirection="column"
        justifyContent="space-around"
      >
        <IconList as="ul" justifyContent="center" alignItems="center">
          <li>
            <a href="http://www.luigiderson.dev">
              <MailIcon size={24} color="green" />
            </a>
          </li>
          <li>
            <a href="http://www.luigiderson.dev">
              <MarkGithubIcon size={23} color="green" />
            </a>
          </li>
        </IconList>

        <Text display="block" fontSize="xs" mb={1}>
          Background by{' '}
          <a
            href="https://unsplash.com/@pawel_czerwinski"
            target="_blank"
            rel="noreferrer norel"
          >
            Paweł Czerwiński
          </a>
        </Text>
      </Footer>
    </>
  )
}

export default ComingSoon

const Section = styled.section`
  height: calc(100vh - ${FOOTER_HEIGHT});
  max-width: 1200px;
  margin: 0 auto;
  text-align: center;
`

const Footer = styled(FlexBox)`
  height: ${FOOTER_HEIGHT};
`

const IconList = styled(FlexBox)`
  list-style: none;
  margin: 0;
  padding: 0;
  & > li {
    padding: 0 ${({ theme }) => theme.space[3]};
  }
`

const BgImg = styled.img`
  width: 100%;
  height: 100%;
  object-fit: cover;
  filter: opacity(0.6);
`
