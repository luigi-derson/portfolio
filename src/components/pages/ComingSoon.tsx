import { MarkGithubIcon, MailIcon, PeopleIcon } from '@primer/styled-octicons'
import { useStaticQuery, graphql } from 'gatsby'

import { Heading } from '@components/atoms/Heading'
import logoSVG from '@src/images/luigi-derson_logo.svg'

import { Text } from '@components/atoms/Text'
import { x } from '@xstyled/styled-components'
import Container from '@components/molecules/Container'

const ComingSoon = () => {
  const data = useStaticQuery(graphql`
    query {
      desktop: file(
        relativePath: { eq: "pawel-czerwinski-GHLapgsw0dg-unsplash.jpg" }
      ) {
        childImageSharp {
          fluid(quality: 90, maxWidth: 1920) {
            ...GatsbyImageSharpFluid_withWebp
          }
        }
      }
    }
  `)

  const imgData = data.desktop.childImageSharp.fluid

  const backgroundFluidImageStack = [
    `url(${imgData.srcWebp})`,
    `linear-gradient(0deg, rgba(0,0,0,0.45) 0%, rgba(0,0,0,0.25) 50%, rgba(0,0,0,0.45) 100%)`,
  ]
  return (
    <x.div
      backgroundImage={backgroundFluidImageStack.join(',')}
      h={1}
      display="flex"
      flexDirection="column"
      backgroundPosition="center"
    >
      <Container textAlign="center" flexGrow={2}>
        <x.div
          display="flex"
          flexDirection="column"
          justifyContent="center"
          alignItems="center"
          h={1}
        >
          <x.div my={4}>
            <x.img h={15} src={logoSVG} alt="Luigi Derson Logo" />
          </x.div>
          <Heading
            as="h2"
            color="#32cd32"
            fontSize={{ _: 'xl', md: '3xl' }}
            fontFamily="mono"
            fontWeight="400"
            mb={4}
          >
            &gt; Comitting changes...
          </Heading>
          <Text as="p" color="white" fontWeight={300}>
            Building a new visual experience
          </Text>
        </x.div>
      </Container>
      <x.div
        as="footer"
        textAlign="center"
        flexDirection="column"
        justifyContent="space-around"
        py={[1, 2, 3]}
      >
        <x.ul
          display="flex"
          justifyContent="center"
          alignItems="center"
          spaceX={4}
        >
          <li>
            <a
              href="mailto:luigi.prodesign@gmail.com"
              target="_blank"
              rel="noreferrer norel"
              title="Send me an email"
            >
              <MailIcon size={24} color="#32cd32" />
            </a>
          </li>
          <li>
            <a
              href="https://github.com/luigi-derson"
              target="_blank"
              rel="noreferrer norel"
              title="Check my github profile"
            >
              <MarkGithubIcon size={23} color="#32cd32" />
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/luigi-sanchez-731b72182/"
              target="_blank"
              rel="noreferrer norel"
              title="Check my Linkedin profile"
            >
              <PeopleIcon size={23} color="#32cd32" />
            </a>
          </li>
        </x.ul>
      </x.div>
      <Text display="block" textAlign="right" fontSize="xs" py="2" pr="3">
        Background by{' '}
        <a
          href="https://unsplash.com/@pawel_czerwinski"
          target="_blank"
          rel="noreferrer norel"
          title="Pawel Czerwinski unsplash profile"
        >
          Paweł Czerwiński
        </a>
      </Text>
    </x.div>
  )
}

export default ComingSoon
