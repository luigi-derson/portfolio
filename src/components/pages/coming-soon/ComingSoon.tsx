import { x } from '@xstyled/styled-components'
import { MarkGithubIcon, MailIcon, PeopleIcon } from '@primer/styled-octicons'
import { StaticImage } from 'gatsby-plugin-image'

import { Heading } from '@components/atoms/Heading'
import { Text } from '@components/atoms/Text'
import { Container } from '@components/molecules/Container'

import logoSVG from './assets/luigi-derson-logo.svg'

export const ComingSoon = () => {
  return (
    <x.div h="calc(100vh - 86.39px)" display="flex" flexDirection="column">
      <StaticImage
        style={{
          position: 'fixed',
          height: '100%',
          width: '100%',
          top: 0,
          left: 0,
          zIndex: -1,
          opacity: 0.65,
          background:
            'linear-gradient(0deg, rgba(0,0,0,0.45) 0%, rgba(0,0,0,0.25) 50%, rgba(0,0,0,0.45) 100%)',
        }}
        src="./assets/pawel-czerwinski-GHLapgsw0dg-unsplash.jpg"
        alt="Pawel Czerwinski Unsplash"
        layout="fullWidth"
        formats={['auto', 'webp', 'avif']}
      />

      <Container textAlign="center" flexGrow={2}>
        <x.div
          display="flex"
          flexDirection="column"
          justifyContent="center"
          alignItems="center"
          h={1}
        >
          <x.div my={4}>
            <x.img
              h={14}
              src={logoSVG}
              height="56px"
              width="56px"
              alt="Luigi Derson Logo"
            />
          </x.div>
          <Heading
            as="h2"
            color="#32cd32"
            fontSize={{ _: 'xl', md: '3xl' }}
            fontFamily="mono"
            fontWeight="400"
            mb={4}
          >
            &gt; Comitting changes...
          </Heading>
          <Text as="p" color="white" fontWeight={300}>
            Building a new visual experience
          </Text>
        </x.div>
      </Container>
      <x.div
        as="footer"
        textAlign="center"
        flexDirection="column"
        justifyContent="space-around"
        py={{ _: 4, md: 6 }}
      >
        <x.ul
          display="flex"
          justifyContent="center"
          alignItems="center"
          spaceX={4}
        >
          <li>
            <a
              href="mailto:luigi.prodesign@gmail.com"
              target="_blank"
              rel="noreferrer norel"
              title="Send me an email"
            >
              <MailIcon size={24} color="#32cd32" />
            </a>
          </li>
          <li>
            <a
              href="https://github.com/luigi-derson"
              target="_blank"
              rel="noreferrer norel"
              title="Check my github profile"
            >
              <MarkGithubIcon size={23} color="#32cd32" />
            </a>
          </li>
          <li>
            <a
              href="https://www.linkedin.com/in/luigi-sanchez-731b72182/"
              target="_blank"
              rel="noreferrer norel"
              title="Check my Linkedin profile"
            >
              <PeopleIcon size={23} color="#32cd32" />
            </a>
          </li>
        </x.ul>
      </x.div>
    </x.div>
  )
}
