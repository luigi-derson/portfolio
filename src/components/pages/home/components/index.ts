export { HeroSection } from './HeroSection'
export { ProjectsSection } from './ProjectsSection'
