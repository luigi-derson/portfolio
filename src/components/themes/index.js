import theme, { colors } from './theme'

// Dark theme
const dark = { ...theme, colors: { ...colors } }

dark.colors.fg = colors.white
dark.colors.bg = colors.black

export { theme as light, dark }
