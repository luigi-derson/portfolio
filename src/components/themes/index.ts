import { defaultTheme, generateHexAlphaVariants, th } from '@xstyled/system'

const fallbackSansFonts = 'ui-sans-serif, system-ui, -apple-system, sans-serif'

const theme = {
  ...defaultTheme,
  fonts: {
    display: `Poppins, ${fallbackSansFonts}`,
    sans: `Inter, ${fallbackSansFonts}`,
    mono: `SFMono-Regular, Menlo, 'Roboto Mono', ui-monospace, Monaco, Consolas, "Liberation Mono", "Courier New", monospace`,
  },
  colors: {
    ...defaultTheme.colors,
    ...generateHexAlphaVariants({
      'dark-blue': '#140a82',
      'dark-red': '#f50a14',
      primary: '#140a82',
      secondary: '#f50a14',
    }),
    text: th.color('true-gray-900-a80'),
    headings: th.color('true-gray-900'),
    background: th.color('gray-50'),
    links: th.color('primary'),
    borders: th.color('true-gray-900'),
    modes: {
      dark: {
        text: th.color('gray-50-a70'),
        background: th.color('true-gray-900'),
        headings: th.color('gray-50'),
        links: th.color('indigo-400'),
        borders: th.color('gray-50'),
      },
    },
  },
  borders: {
    thin: '2px solid',
  },
}

export default theme
