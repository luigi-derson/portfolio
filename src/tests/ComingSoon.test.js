import renderer from 'react-test-renderer'
import ComingSoon from 'components/pages/ComingSoon'

import { ThemeProvider } from 'styled-components'
import { light } from 'components/themes'

test('renders correctly', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={light}>
        <ComingSoon />
      </ThemeProvider>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
