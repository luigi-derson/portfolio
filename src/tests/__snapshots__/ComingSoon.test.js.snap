// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="sc-hcevGk hlyZmQ"
  >
    <img
      className="sc-cfZGEC hMegRr"
      src="pawel-czerwinski-GHLapgsw0dg-unsplash-optimized.jpg"
    />
  </div>,
  <section
    className="sc-koaBLD brSmmN"
  >
    <div
      className="sc-hcevGk sc-oHXjo bEBkpf kuuuFX"
    >
      <div
        className="sc-hcevGk cIDvdt"
      >
        <img
          alt="Luigi Derson Logo"
          height="60px"
          src="luigi-derson_logo.svg"
        />
      </div>
      <h2
        className="sc-dYzljZ bhbwzu"
      >
        &gt; Comitting changes...
      </h2>
      <p
        className="sc-cRcunm"
      >
        Building a new visual experience
      </p>
    </div>
  </section>,
  <footer
    className="sc-hcevGk sc-oHXjo sc-kTaSZA bHVYYe fJkfGG eQnJUa"
  >
    <ul
      className="sc-hcevGk sc-oHXjo sc-ha-DqYV  ejnswD eMrWzr"
    >
      <li>
        <a
          href="http://www.luigiderson.dev"
        >
          <svg
            aria-hidden="true"
            className="sc-eWVKcp jepRF"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M1.75 3A1.75 1.75 0 000 4.75v14c0 .966.784 1.75 1.75 1.75h20.5A1.75 1.75 0 0024 18.75v-14A1.75 1.75 0 0022.25 3H1.75zM1.5 4.75a.25.25 0 01.25-.25h20.5a.25.25 0 01.25.25v.852l-10.36 7a.25.25 0 01-.28 0l-10.36-7V4.75zm0 2.662V18.75c0 .138.112.25.25.25h20.5a.25.25 0 00.25-.25V7.412l-9.52 6.433c-.592.4-1.368.4-1.96 0L1.5 7.412z\\"></path>",
              }
            }
            fill="currentColor"
            height={24}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 24 24"
            width={24}
          />
        </a>
      </li>
      <li>
        <a
          href="http://www.luigiderson.dev"
        >
          <svg
            aria-hidden="true"
            className="sc-iGctRS UTlxZ"
            dangerouslySetInnerHTML={
              Object {
                "__html": "<path fill-rule=\\"evenodd\\" d=\\"M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z\\"></path>",
              }
            }
            fill="currentColor"
            height={23}
            role="img"
            style={
              Object {
                "display": "inline-block",
                "userSelect": "none",
                "verticalAlign": "text-bottom",
              }
            }
            viewBox="0 0 16 16"
            width={23}
          />
        </a>
      </li>
    </ul>
    <span
      className="sc-hLGeHF tdnso"
      display="block"
      fontSize="xs"
    >
      Background by
       
      <a
        href="https://unsplash.com/@pawel_czerwinski"
        rel="noreferrer norel"
        target="_blank"
      >
        Paweł Czerwiński
      </a>
    </span>
  </footer>,
]
`;
